### PR TITLE
Improved CMake support for cross-compilation with MinGW

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull requests resolves some issues with using CMake to cross-compile LAMMPS executables for Windows. It also enables support for LATTE and includes now the option to build with MPI enabled and to use the ninja-build for faster, parallel compilation.

**Author(s)**

Axel Kohlmeyer (Temple U)  with significant inspiration from PR #1940 by @junghans 

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Implementation Notes**

Precompiled MPI binary files are required and this is now configured as an external project with download of the required package from the server at lammps.org.

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [ ] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system

### CODE REVIEW

File: cmake/Modules/Packages/KIM.cmake [hunk 1]
```diff
@@ -49,11 +51,12 @@ if(PKG_KIM)
                  -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
                  -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
                  -DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}
-      BUILD_BYPRODUCTS <INSTALL_DIR>/${CMAKE_INSTALL_LIBDIR}/libkim-api${CMAKE_SHARED_LIBRARY_SUFFIX} 
+                 -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                 BUILD_BYPRODUCTS <INSTALL_DIR>/${_KIM_LIBDIR}/libkim-api${CMAKE_SHARED_LIBRARY_SUFFIX}
```
Comment (junghans) [inline cmake/Modules/Packages/KIM.cmake]: Indentation...

File: cmake/Modules/Packages/KIM.cmake [hunk 2]
```diff
@@ -35,6 +35,8 @@ if(PKG_KIM)
   option(DOWNLOAD_KIM "Download KIM-API from OpenKIM instead of using an already installed one" ${DOWNLOAD_KIM_DEFAULT})
   if(DOWNLOAD_KIM)
     message(STATUS "KIM-API download requested - we will build our own")
+    # Workaround for cross compilation with MinGW, which messes up ${CMAKE_INSTALL_LIBDIR}
+    string(REPLACE ${CMAKE_INSTALL_PREFIX} "" _KIM_LIBDIR ${CMAKE_INSTALL_LIBDIR})
```
Comment (junghans) [inline cmake/Modules/Packages/KIM.cmake]: So this basically takes the `/lib` port of `${CMAKE_INSTALL_LIBDIR}`, correct? The current comment is not very helpful....

Isn't `${CMAKE_INSTALL_LIBDIR}` just `lib` (or `lib64`) usually? And `${CMAKE_INSTALL_FULL_LIBDIR}` the full path?

File: cmake/CMakeLists.txt [hunk 3]
```diff
@@ -255,8 +262,8 @@ if(PKG_MSCG OR PKG_USER-ATC OR PKG_USER-AWPMD OR PKG_USER-QUIP OR PKG_LATTE)
     enable_language(Fortran)
     file(GLOB LAPACK_SOURCES ${LAMMPS_LIB_SOURCE_DIR}/linalg/[^.]*.[fF])
     add_library(linalg STATIC ${LAPACK_SOURCES})
-    set(BLAS_LIBRARIES linalg)
-    set(LAPACK_LIBRARIES linalg)
+    set(BLAS_LIBRARIES "${CMAKE_BINARY_DIR}/liblinalg.a")
+    set(LAPACK_LIBRARIES "${CMAKE_BINARY_DIR}/liblinalg.a")
```
Comment (junghans) [inline cmake/CMakeLists.txt]: That feel wrong...why is that needed?

File: cmake/Modules/Packages/LATTE.cmake [hunk 4]
```diff
@@ -18,17 +20,20 @@ if(PKG_LATTE)
       -DBLAS_LIBRARIES=${BLAS_LIBRARIES} -DLAPACK_LIBRARIES=${LAPACK_LIBRARIES}
```
Comment (junghans) [inline cmake/Modules/Packages/LATTE.cmake]: I guess set should look up the location here, instead of doing the hack above, something like:
```
get_target_property(BLAS_LOCATION linalg LOCATION)
```

File: cmake/Modules/Packages/KIM.cmake (hunk 2 above)
Comment (akohlmey) [reply cmake/Modules/Packages/KIM.cmake]: not when cross-compiling

File: cmake/CMakeLists.txt (hunk 3 above)
Comment (akohlmey) [reply cmake/CMakeLists.txt]: because LATTE won't find it otherwise and thus configuration of LATTE fails unless there is a publicly installed BLAS/LAPACK library.

File: cmake/Modules/Packages/KIM.cmake (hunk 2 above)
Comment (junghans) [reply cmake/Modules/Packages/KIM.cmake]: I guess a better comment would be something like: "When cross-compilation (e.g. MinGW) CMAKE_INSTALL_LIBDIR contains the absolute path, hence with strip the libdir portion".

File: cmake/CMakeLists.txt (hunk 3 above)
Comment (junghans) [reply cmake/CMakeLists.txt]: On a second though that is ok, just add a comment.

Comment (junghans) [reply cmake/CMakeLists.txt]: we might still want to use:
```
get_target_property(BLAS_LIBRARIES linalg LOCATION)
get_target_property(LAPACK_LIBRARIES linalg LOCATION)
```
instead.

File: cmake/Modules/Packages/KIM.cmake (hunk 2 above)
Comment (junghans) [reply cmake/Modules/Packages/KIM.cmake]: Also we might just want to do that once instead of here and in `LATTE.cmake`.

File: cmake/Modules/Packages/LATTE.cmake (hunk 4 above)
Comment (akohlmey) [reply cmake/Modules/Packages/LATTE.cmake]: this gives an error.

File: cmake/CMakeLists.txt (hunk 3 above)
Comment (junghans) [reply cmake/CMakeLists.txt]: Even though merged we should still do this.

Comment (akohlmey) [reply cmake/CMakeLists.txt]: Please figure it out and submit a pull request. I have googled until my fingers started to bleed, but I either get an error message from cmake telling me that I should not to this, or did not get the answer I am looking for. This works well enough and I need the time to complete other open PRs by tonight.

Comment (junghans) [reply cmake/CMakeLists.txt]: CMake's error message basically told me what to do: see #1950.
